src/replication/adapter/dtr/client.go

```diff
@@ -0,0 +1,331 @@
+			query := urlAPI.Query()
+			query.Set("pageStart", nextPage)
+			endpoint = urlAPI.Scheme + "://" + urlAPI.Host + urlAPI.Path + "?" + query.Encode()
+		}
+	}
+	rv.Elem().Set(resources)
+	return nil
+}
+
+// getRepositories returns a list of repositories in DTR
+func (c *Client) getRepositories() ([]*adp.Repository, error) {
+	var repositories []Repository
+	var dtrRepositories Repositories
+
+	endpoint := fmt.Sprintf("%s/api/v0/repositories?pageSize=100", c.url)
+	urlAPI, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	for len(endpoint) > 0 {
+		req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+		if err != nil {
+			return nil, err
+		}
+		req.SetBasicAuth(c.username, c.password)
+		resp, err := c.client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		if resp.StatusCode < 200 || resp.StatusCode > 299 {
+			return nil, &common_http.Error{
+				Code:    resp.StatusCode,
+				Message: string(data),
+			}
+		}
+
+		if err = json.Unmarshal(data, &dtrRepositories); err != nil {
+			log.Errorf("Failed to parse json response")
+			log.Errorf("%v", err)
+			log.Errorf("%s", string(data))
+
+			return nil, err
+		}
+
+		// merge the arrays
+		repositories = append(repositories, dtrRepositories.Repositories...)
+		endpoint = ""
+
+		nextPage := resp.Header.Get("X-Next-Page-Start")
+		if len(nextPage) > 0 {
+			query := urlAPI.Query()
+			query.Set("pageStart", nextPage)
+			endpoint = urlAPI.Scheme + "://" + urlAPI.Host + urlAPI.Path + "?" + query.Encode()
+		}
+	}
+
+	result := []*adp.Repository{}
+	for _, repository := range repositories {
+		log.Debugf("Processing DTR repo %s", repository.Name)
+		result = append(result, &adp.Repository{
+			ResourceType: string(model.ResourceTypeImage),
+			Name:         fmt.Sprintf("%s/%s", repository.Namespace, repository.Name),
+		})
+	}
+	return result, nil
+}
+
+// getVTags looks up a repositories tags in DTR
+func (c *Client) getVTags(repository string) ([]*adp.VTag, error) {
+	var tags []*Tag
+	// This assumes repository is of form namespace/repo
+	urlAPI := fmt.Sprintf("%s/api/v0/repositories/%s/tags?pageSize=100", c.url, repository)
+	log.Debugf("Looking up tags for %s at %s", repository, urlAPI)
+	if err := c.getAndIteratePagination(urlAPI, &tags); err != nil {
+		log.Debugf("Failed looking up tags for %s at %s", repository, urlAPI)
+		return nil, err
+	}
+
+	var result []*adp.VTag
+	for _, tag := range tags {
+		result = append(result, &adp.VTag{
+			ResourceType: string(model.ResourceTypeImage),
+			Name:         tag.Name,
+		})
+	}
+	return result, nil
+}
+
+// getNamespaces returns DTR namespaces.  DTR also calles these orgs and accounts depending on where you look
+func (c *Client) getNamespaces() ([]Account, error) {
+	var accounts []Account
+	var response Accounts
+
+	endpoint := fmt.Sprintf("%s/enzi/v0/accounts?limit=100", c.url)
```

Comment:
Can you share the api document that you are referenced here

Reply:
https://docs.docker.com/v17.09/datacenter/dtr/2.3/reference/api/

Unfortunately this API is no longer documented in the latest API docs however it is still there.  It is the only way to create orgs as the core DTR api does not support any of those features based on how tied it is to UCP.